### PR TITLE
[ESP32] - Make sure that BLE indications are sent sequentially

### LINKED ISSUE #22444: [1.0] ESP32 - BLE Indications should be sequential
### Reproduction steps

```bash
It can happen that when large amount of data is sent through BLE indications (with lesser MTU, may be) this can cause problems of resource exhaustion at the BLE layer. As indications are acked, the next should be sent once the ack for the previous one is received.

This may be related: https://github.com/project-chip/connectedhomeip/issues/22416 
cc @yufengwangca
```

### Bug prevalence

Not everytime. Can occur during stress tests

### GitHub hash of the SDK that was being used

33f6a910cd9a8a0cfdd7088e2f43efd2f7f566a7

### Platform

esp32

### Platform Version(s)

_No response_

### Type

Platform Issue

### Anything else?

As per the BLE specification:
```
Indications sent from a server also use a sequential indication-confirmation
protocol. No other indications shall be sent to the same client from this server
until a confirmation PDU has been received.
```

### PR DESCRIPTION
**Pull request by dhrishi**
The next indication is sent only after the previous one is acked

#### Problem
* Fixes #22444

@yufengwangca As per our offline discussion, please try it for your automated test issue reported in #22416 

#### Change overview
A status of 0 indicates that the indication is sent from the BLE host. Ideally, we should move to the next step when the status received is BLE_HS_EDONE which indicates the indication ack is received. For any other status value, give an appropriate error.

In short, handle all the status values except for 0.

#### Testing
* BLE based commissioning successful

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by yufengwangca** [comment]
I stress this this patch yesterday and didn't see the problem, but I am not sure if this patch could address the original issue we hit since it is very hard to reproduce.

From log, it shows the issue is in BLEManagerImpl::SendIndication() and receive error BLE_HS_ENOMEM, this error is still handled as an error with this patch.

**Review by woody-apple — APPROVED** [review]

**Comment by dhrishi** [comment]
>From log, it shows the issue is in BLEManagerImpl::SendIndication() and receive error BLE_HS_ENOMEM, this error is still handled as an error with this patch.

Thanks for running the test @yufengwangca. There is a chance that you received no memory error as the previous indications were not acked and they still consumed some memory, while the new one was sent 

IMO, even if the issue you encountered is related to something else, this fix is still required :-)
